src/tiled/objectsdock.h

```diff
@@ -57,23 +58,32 @@ private slots:
     void documentAboutToClose(Document *document);
     void moveObjectsUp();
     void moveObjectsDown();
+    void filterObjects();
 
 private:
     void retranslateUi();
 
-    void saveExpandedGroups();
-    void restoreExpandedGroups();
+    void saveExpandedGroups(QMap<MapDocument*, QList<ObjectGroup*> > &expansionState);
+    void restoreExpandedGroups(QMap<MapDocument*, QList<ObjectGroup*> > &expansionState);
 
     QAction *mActionNewLayer;
     QAction *mActionObjectProperties;
     QAction *mActionMoveToGroup;
     QAction *mActionMoveUp;
     QAction *mActionMoveDown;
 
+    QLineEdit *mFilterEdit;
     ObjectsView *mObjectsView;
     MapDocument *mMapDocument;
+
+    // These maps store the state of the objects dock for each opened map document
     QMap<MapDocument*, QList<ObjectGroup*> > mExpandedGroups;
+    QMap<MapDocument*, QList<ObjectGroup*> > mPrefilterExpandedGroups;
+    QMap<MapDocument*, QString> mFilterStrings;
+    QMap<MapDocument*, bool> mFilterWasEmpty;
+
     QMenu *mMoveToMenu;
+    QModelIndex getGroupIndex(ObjectGroup *og);
```

Comment:
I'm not sure what you mean by convenience override.

But here is what I did so far, I moved the getGroupIndex function from the objects dock to the objects view (it uses mapToViewModel for conversion), then made it private and made public helper functions in the objects view(isGroupExpanded, setGroupExpanded), so that the objectsDock can use them without worrying about the conversion.